File: GenesisAndroid/src/test/java/com/emerchantpay/gateway/genesisandroid/TransactionUnitTest.java
package com.emerchantpay.gateway.genesisandroid;

import com.emerchantpay.gateway.genesisandroid.api.models.Transaction;
import com.emerchantpay.gateway.genesisandroid.api.util.NodeWrapper;

import org.junit.Before;
import org.junit.Test;

import java.math.BigDecimal;

import static junit.framework.Assert.assertEquals;
import static org.mockito.Mockito.mock;
import static org.mockito.Mockito.when;

public class TransactionUnitTest {

    private NodeWrapper wrapper;
    private Transaction transaction;

    @Before
    public void setup() {
        wrapper = mock(NodeWrapper.class);
    }

    @Test
    public void testTransaction() {
        when(wrapper.findString("status")).thenReturn("approved");
        when(wrapper.findBigDecimal("amount")).thenReturn(new BigDecimal("2.00"));

        transaction = new Transaction(wrapper);

        assertEquals(transaction.getStatus(), wrapper.findString("status"));
        assertEquals(transaction.getAmount(), wrapper.findBigDecimal("amount"));
    }
}

File: GenesisAndroid/src/test/java/com/emerchantpay/gateway/genesisandroid/TransactionUnitTest.kt
package com.emerchantpay.gateway.genesisandroid

import com.emerchantpay.gateway.genesisandroid.api.models.Transaction
import com.emerchantpay.gateway.genesisandroid.api.util.NodeWrapper

import org.junit.Before
import org.junit.Test

import java.math.BigDecimal

import junit.framework.Assert.assertEquals
import org.mockito.Mockito.mock
import org.mockito.Mockito.`when`

class TransactionUnitTest {

    private var wrapper: NodeWrapper? = null
    private var transaction: Transaction? = null

    @Before
    fun setup() {
        wrapper = mock(NodeWrapper::class.java)
    }

    @Test
    fun testTransaction() {
        `when`(wrapper!!.findString("status")).thenReturn("approved")
        `when`(wrapper!!.findBigDecimal("amount")).thenReturn(BigDecimal("2.00"))

        transaction = Transaction(wrapper)

        assertEquals(transaction!!.status, wrapper!!.findString("status"))
        assertEquals(transaction!!.amount, wrapper!!.findBigDecimal("amount"))
    }
}
